tensorflow/python/data/util/structure_test.py

```diff
@@ -748,6 +749,17 @@ def testCustomMapping(self):
     self.assertIsInstance(spec, CustomMap)
     self.assertEqual(spec["foo"], tensor_spec.TensorSpec([], dtypes.float32))
 
+  def testAttrClass(self):
+    @attr.s
+    class AttrClass:
+        x = attr.ib()
+        y = attr.ib()
+
+    elem = AttrClass(x=constant_op.constant(1.), y=constant_op.constant(2.))
+    spec = structure.type_spec_from_value(elem)
+    self.assertIsInstance(spec, AttrClass)
+    self.assertEqual(spec.x, tensor_spec.TensorSpec([], dtypes.float32))
+
```

Comment:
Please add a test for an `ObjectProxy` of `attr.s`, similar to the test for `ObjectProxy` of `namedtuple` below